# HG changeset patch
# User Paul Boddie <paul@boddie.org.uk>
# Date 1359762149 -3600
# Node ID f192339c4473f140b07a1522ecac636abe616350
# Parent  f548add9c9d5a0c445c752c82fb6465c2b442d21
Fixed the plain text formatting of links.

diff -r f548add9c9d5 -r f192339c4473 MoinMoin/formatter/text_plain.py
--- a/MoinMoin/formatter/text_plain.py	Fri Feb 01 01:38:47 2013 +0100
+++ b/MoinMoin/formatter/text_plain.py	Sat Feb 02 00:42:29 2013 +0100
@@ -45,14 +45,14 @@
             self._text = []
             return u''
         else:
-            if "".join(self._text) == self._url:
+            try:
+                if "".join(self._text) == self._url:
+                    return ''
+                else:
+                    return u' [%s]' % self._url
+            finally:
                 self._url = None
                 self._text = None
-                return ''
-            else:
-                self._url = None
-                self._text = None
-                return u' [%s]' % (self._url)
 
     def url(self, on, url='', css=None, **kw):
         if on:
@@ -60,14 +60,14 @@
             self._text = []
             return u''
         else:
-            if "".join(self._text) == self._url:
+            try:
+                if "".join(self._text) == self._url:
+                    return ''
+                else:
+                    return u' [%s]' % self._url
+            finally:
                 self._url = None
                 self._text = None
-                return ''
-            else:
-                self._url = None
-                self._text = None
-                return u' [%s]' % (self._url)
 
     def attachment_link(self, on, url=None, **kw):
         if on:
